crates/evm/src/query.rs

```diff
@@ -616,59 +584,51 @@ impl<C: sov_modules_api::Context> Evm<C> {
     ) -> RpcResult<AccessListWithGasUsed> {
         let mut request = request.clone();
 
-        let (l1_fee_rate, mut block_env, block_num) = {
-            let block = match self.get_sealed_block_by_number(block_number, working_set)? {
-                Some(block) => block,
-                None => return Err(EthApiError::UnknownBlockNumber.into()),
-            };
-            let l1_fee_rate = block.l1_fee_rate;
+        let (l1_fee_rate, block_env, mut cfg_env) = {
+            let block = self
+                .get_sealed_block_by_number(block_number, working_set)?
+                .ok_or(EthApiError::UnknownBlockNumber)?;
             let block_env = BlockEnv::from(&block);
 
-            (l1_fee_rate, block_env, block.header.number)
+            let cfg = self
+                .cfg
+                .get(working_set)
+                .expect("EVM chain config should be set");
```

Comment:
reading config should happen after setting state to some block.

imagine you make a fork in the future, and change some value in config on the activation block.

with this setup, you are gonna run the latest cfg on a block from the past, which is possibly on an older version of the chain.

now your simulation will be wrong.

Reply:
fixed